### ISSUE TITLE
add example model and testthat calls to ensure package is up-to-date

### ISSUE THREAD

**Issue by chantelwetzel-noaa** [nwfsc-assess/nwfscDiag#11]
To ensure continued compatibility between the package, r4ss, and current SS3 versions need to create tests to ensure there are no revisions in any of the packages used that breaks the nwfscDiag package.

**Comment by chantelwetzel-noaa** [comment]
I think that is a good example model to use. However, I am not entirely clear on the most streamlined way to create a test via github-actions since I have not set this up before but can look some examples.

**Comment by kellijohnson-NOAA** [comment]
@k-doering-NOAA is our resident GitHub actions master!

**Comment by k-doering-NOAA** [comment]
Great idea @chantelwetzel-noaa ! R4ss has similar test currently, and uses this GitHub action workflow to download SS3 and the test-models repo before running https://github.com/nmfs-stock-synthesis/workflows/blob/e596010cf5f015aa4982d54c2b391de5bd6aa0cb/.github/workflows/r4ss-extra-tests.yml .

We could probably augment this to work for both this pig and r4ss.

An example of tests that need to run SS3 is here: https://github.com/r4ss/r4ss/blob/82abd14df22bc6ad39fb9b59e371d2e16944a805/tests/testthat/test-runs.R 

Basically, you just need to skip them if the exe is not available. 

Anyway, more than happy to help set this up!

**Comment by kellijohnson-NOAA** [comment]
@chantelwetzel-noaa sorry for all the small commits to main. I was trying to get the GitHub action to work. Some of what I tried was dumb, other things have been helpful. It is now failing on the install of {stringr} which was just updated yesterday on CRAN. I tried switching to the newest version of Ubuntu and that did not help. I will keep trying some things.

**Comment by chantelwetzel-noaa** [comment]
Thanks for working on this. It started failing when I change the name name to main (I had similar issues with the nwfscSurvey) package and was not able to figure out the issue.

**Comment by k-doering-NOAA** [comment]
I think the failing gha might be related to some recent updates to r cmd check workflows, here is how I fixed it in ghactions4r, which I figured out from looking at r-lib examples: https://github.com/nmfs-fish-tools/ghactions4r/blob/main/.github/workflows/r-cmd-check.yml

Also, one longer term fix would be to reference the workflow in ghactions4r instead, so when updates are pushed to that package, your workflows automatically will be updated, also! Instruction on how to do that are in the [ghactions4r readme](https://github.com/nmfs-fish-tools/ghactions4r#how-do-i-use-these-workflows-in-my-r-package)

**Comment by kellijohnson-NOAA** [comment]
Thanks @k-doering-NOAA for looking into this. You were right about the environment variables. Given that we need SS3 exe file we have to have our own workflow but I simplified it as you suggested 🥇 . @chantelwetzel-noaa I think we can close this issue now given that the tests are passing with the model run. Let me know if there is something additional we need to do.

**Comment by chantelwetzel-noaa** [comment]
@k-doering-NOAA @kellijohnson-NOAA Thank you!
